Content.Client/Doors/Electronics/DoorElectronicsBoundUserInterface.cs

```diff
@@ -22,7 +22,7 @@ protected override void Open()
 
         foreach (var accessLevel in _prototypeManager.EnumeratePrototypes<AccessLevelPrototype>())
         {
-            if (accessLevel.Name != null)
+            if (accessLevel.Name != null && accessLevel.CanDoorElectronicsConfiguratorChange == true)  //SS220 DoorElectronicsConfiguratorFix
```

Comment:
` accessLevel.CanDoorElectronicsConfiguratorChange == true`
У тебя `CanDoorElectronicsConfiguratorChang` уже и так переменная типа bool сравнивать с true излишне.
Также нейминг `CanDoorElectronicsConfiguratorChang` сбивает с толку, а в контексте того, что это переменная AccessLevelPrototype уже можно и сильно призадуматься.
 Лучше бы подошло название CanBeSwitchableInRound. К Switchable тоже есть вопросы, но adjective лучше не нашёл. 
 Отличается более подробным описанием что с этим доступом делается и не ссылается на некий "DoorElectronicsConfigurator", тем более что речь про доступы, и не про какие DoorElectronic и тд мы не знаем.
 Ну и Chang тоже неудачно, что значит доступ меняется в контексте самого себя?
 Душный ли я? Определёно :clueless:, но всё-таки нам работать в этом проекте долго, поэтому нужно следить за внятностью написанного в контексте человека, который впервые видит это. 
```suggestion
            if (accessLevel.Name != null && accessLevel.CanDoorElectronicsConfiguratorChange)  //SS220 DoorElectronicsConfiguratorFix
```

Reply:
Мультитул и конфигуратор сетей имеют тег `DoorElectronicsConfigurator`, который и отвечает за вызов UI по смене доступов, при использовании на предмет, имеющий этот же тег (микросхема шлюза в данном случае) , поэтому я и назвал её так

Reply:
`CanBeSwitchableInRound` также не оч. подходит из-за того, что эти доступы можно менять в раунде при помощи конфигуратора доступов (даже на плате шлюза), а фикс направлен именно на то, чтобы эти доступы нельзя было изменить при помощи мультитула или конфигуратора сетей......

Reply:
> Мультитул и конфигуратор сетей имеют тег `DoorElectronicsConfigurator`, который и отвечает за вызов UI по смене доступов, при использовании на предмет, имеющий этот же тег (микросхема шлюза в данном случае) , поэтому я и назвал её так

Вернее сама микросхема шлюза для открытия UI по смене доступов запрашивает предмет с тегом `DoorElectronicsConfigurator`

Reply:
> `CanBeSwitchableInRound` также не оч. подходит из-за того, что эти доступы можно менять в раунде при помощи конфигуратора доступов (даже на плате шлюза), а фикс направлен именно на то, чтобы эти доступы нельзя было изменить при помощи мультитула или конфигуратора сетей......

Да, правда твоя, у нас эти же доступы и через хопа можно ставить, тогда надо уточнить, что речь про чтение этого доступа: `CanBeSwitchableInReader` чей reader вроде понятно, мы живём в accesLevel, но можно и дописать. Других способов лезть в AccessReader дверей кроме конфигуратора или мультитула я не помню. В раунде так точно. Тем более в целом мы и через консоль запрещаем с ними взаимодействовать, поэтому общность не мешает. 

> Мультитул и конфигуратор сетей имеют тег `DoorElectronicsConfigurator`, который и отвечает за вызов UI по смене доступов, при использовании на предмет, имеющий этот же тег (микросхема шлюза в данном случае) , поэтому я и назвал её так

А если тег будет по другому назван, то менять название и этой перменной? А если у нас будет ивент, который рандомит доступы, то снова делать свою переменную и для него? Ты создаёшь переменную в accesLevel и должно быть понятно, что она делает без дополнительных поисков по коду каких-то тегов.

Если уж не нравится адаптация, то можно добавить enum описывающий свойства доступа, как например ЦК-шный он или нет, является ли он HighSecZone и тд.

Reply:
> > `CanBeSwitchableInRound` также не оч. подходит из-за того, что эти доступы можно менять в раунде при помощи конфигуратора доступов (даже на плате шлюза), а фикс направлен именно на то, чтобы эти доступы нельзя было изменить при помощи мультитула или конфигуратора сетей......
> 
> Да, правда твоя, у нас эти же доступы и через хопа можно ставить, тогда надо уточнить, что речь про чтение этого доступа: `CanBeSwitchableInReader` чей reader вроде понятно, мы живём в accesLevel, но можно и дописать. Других способов лезть в AccessReader дверей кроме конфигуратора или мультитула я не помню. В раунде так точно. Тем более в целом мы и через консоль запрещаем с ними взаимодействовать, поэтому общность не мешает.
> 
> > Мультитул и конфигуратор сетей имеют тег `DoorElectronicsConfigurator`, который и отвечает за вызов UI по смене доступов, при использовании на предмет, имеющий этот же тег (микросхема шлюза в данном случае) , поэтому я и назвал её так
> 
> А если тег будет по другому назван, то менять название и этой перменной? А если у нас будет ивент, который рандомит доступы, то снова делать свою переменную и для него? Ты создаёшь переменную в accesLevel и должно быть понятно, что она делает без дополнительных поисков по коду каких-то тегов.
> 
> Если уж не нравится адаптация, то можно добавить enum описывающий свойства доступа, как например ЦК-шный он или нет, является ли он HighSecZone и тд.

Та не, я не против переименования, я просто объяснил свой выбор именно этого названия)
Поменяю на `CanBeSwitchableInReader`, согласен с тем что это будет понятнее, чем искать тег в прототипах